main.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'nameable'
+
+# Base decorator
+class Decorator < Nameable
```

Comment:
- Hi, Please, let's add the setter/getter for the nameable to make it accessible `attr_accessor :nameable`